### ISSUE TITLE
"On-the-fly mask change" doesnt work properly

### ISSUE THREAD

**Issue by ppalmeida** [igorescobar/jQuery-Mask-Plugin#41]
Hello.
First of all, thank you so much. Thank you for take time to make this excelent plugin. Really like it.

Im trying to do the "ninth digit" (sp nono dígito) mask with "on-the-fly mask change". 

Here is my code:

``` javascript

// $('#num_telephone') reffers to an input text field
var masks = ['9999.9999', '9.9999.9999'];
var options = {
            onKeyPress: function(telephone) {
                var mask = (telephone.length < 9) ? masks[0] : masks[1];
                $('#num_telephone').mask(mask, this);
            }
        };
        $('#num_telephone').mask(masks[0], options);
```

If i try to digit 123456789, until '7' it is ok, keep the 9999.9999 mask. But as soon i insert the '8' digit (should change the mask) the value of the field should be '1234.5678' but it shows '123.4', although the "maxlength" in the input tag changes to "11" as expected.

I suspect the code is updating the value just with the first part of the mask (like '1234.5678' takes just '1234').

So, in summary:
Masks should change from '1234.5678' to '1.2345.6789' but it renders '1.234'.

Thank u again and keep up the good work.

Pedro Paulo Almeida
São Paulo, Brazil

**Comment by Liioo** [comment]
**This is not working anymore**(at least for me).
Tried to test the example of @ppalmeida , http://jsfiddle.net/sA2xn/ and it always return the first mask, no changing and blocking further typing.
Other example that is not working: https://jsfiddle.net/Lioo_/e0w2c6kL/
